backend/modules/pdf_processing/progress_service.py

```
@@ -0,0 +1,235 @@
+        
+        # Send update to frontend
+        asyncio.create_task(self.send_progress(session_id, {
+            "type": "progress_update",
+            "session_id": session_id,
+            "overall_progress": progress_info["overall_progress"],
+            "current_stage": stage,
+            "stage_progress": progress,
+            "message": overall_message,  # Use overall message instead of stage-specific message
+            "details": details or {},
+            "timestamp": time.time()
+        }))
```

Ruff (0.14.5)

Store a reference to the return value of `asyncio.create_task`

(RUF006)